### ISSUE TITLE
Some classes have "→ None" in their Sphinx-generated docs

### ISSUE THREAD

**Issue by sorcio** [python-trio/trio#529]
Some examples:

`class trio.Event → None`
`class trio.Lock → None`
`class trio.StrictFIFOLock → None`
`class trio.StapledStream(send_stream, receive_stream) → None`
`class trio.testing.Sequencer → None`
`class trio.hazmat.ParkingLot → None`

At first glance it looks like an artifact of @attr.s classes. Maybe attrs sets a return annotation somewhere and Sphinx/docutils picks it up?

**Comment by RazerM** [comment]
attrs is [adding an annotation to the generated `__init__`](https://github.com/python-attrs/attrs/commit/fc033b41a11c9ef3e5fd7bee7f936427d4bb5f3d#diff-21756223332b04173cec8910fda6655bR1264), which is picked up by autodoc

**Comment by smurfix** [comment]
Hmm, the annotation is technically correct – I'd teach autodoc to special-case __init__ to not print the return value.

**Comment by njsmith** [comment]
Yeah, autodoc definitely shouldn't be printing return annotations on class constructors – the annotation is correct for `__init__`, but Sphinx is rendering it like it's the annotation for `type.__call__`, and it certainly isn't correct there :-)

**Comment by sorcio** [comment]
Sphinx is planning to fix this for 1.7.5 (which is due in a couple weeks) so I guess we are not going to work around it in Trio.
